### PR TITLE
Fix Marshal Com Helpers

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
I have somehow missed these when cleaning up the function pointer intrinsics.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Com.cs [hunk 1]
```diff
@@ -23,8 +23,7 @@ public static unsafe int AddRef(IntPtr pUnk)
             if (pUnk == IntPtr.Zero)
                 throw new ArgumentNullException(nameof(pUnk));
 
-            return CalliIntrinsics.StdCall__AddRef(((__com_IUnknown*)(void*)pUnk)->pVtable->
-                pfnAddRef, pUnk);
+            return ((delegate* unmanaged<IntPtr, int>)(*(*(void***)pUnk + 1 /* IUnknown.AddRef slot */)))(pUnk);
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Com.cs:26]: Copy & paste from CoreCLR file

Comment (MichalStrehovsky) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Com.cs:26]: Yes, I remember this line being popular on twitter.

**Review by MichalStrehovsky — APPROVED** [review]
Weird that they didn't have McgIntrinsicsAttribute.

**Comment by jkotas** [comment]
It had McgIntrinsicsAttribute via partial class.

**Comment by MichalStrehovsky** [comment]
> It had McgIntrinsicsAttribute via partial class.

Ah. No wonder this was missed. Good riddance.

**Pull request merged**